docs/network-binding-plugin.md

```diff
@@ -0,0 +1,342 @@
+    }, nil
+}
+```
+
+It is recommended to use the latest available hook version.
+The unused methods must be defined to satisfy the interface, preferably with
+a default non-error response.
+
+#### Pod interface name
+
+The name of the network interface in the pod, to which the relevant network
+is to bind to, is expected to be known.
+
+- For the primary network, the default `eth0` naming is expected.
+- For the secondary networks, Kubevirt explicitly specifies the name through
+  Multus annotations of the pod. By that it assumes the CNI plugin used
+  confirms with the CNI specification and will name the interface accordingly.
+  The name used is generated by Kubevirt with the following format:
+  `pod<network name hash ID>`.
+  - Network hash ID: The hash ID is generated based on the network name specified
+    in the VM/VMI interface & network spec.
+    To programmatically generate it, one can use the following code snippet:
+
+```go
+import "kubevirt.io/kubevirt/pkg/network/namescheme"
+
+ifaceName := namescheme.GenerateHashedInterfaceName(network_name)
+```
+
+#### Domain configuration
+
+Changing the domain configuration usually involves the addition of a device
+or a change in an existing device.
+
+Either way, it is important to understand which are the networking
+devices and what are the expectations from Kubevirt:
+
+- Domain API: "kubevirt.io/kubevirt/pkg/virt-launcher/virtwrap/api"
+  - `DomainSpec`: The domain configuration spec structure, which is in sync
+    with [libvirt domain configuration](https://libvirt.org/formatdomain.html).
+    - `Devices`:
+      - `Interface`: The network interface configuration spec.
+      - `HostDevice`: The general host device configuration spec.
+        The host-device is used to pass-through devices from the host to the domain.
+        SR-IOV virtual functions are a classic example for networking.
+- Device Alias:
+  Kubevirt marks on the domain device spec an alias that references
+  the logical network name as appearing in the VM/VMI spec.
+  - For SR-IOV devices, the alias also includes a prefix: `sriov-`.
+
+  ```go
+  import domainschema "kubevirt.io/kubevirt/pkg/virt-launcher/virtwrap/api"
+  deviceAlias := domainschema.NewUserDefinedAlias("sriov-" + "mynet1")
+  ```
+- Parameters: Several configuration parameters of an interface from the VMI spec
+  are passed to the domain device configuration directly (e.g. the mac address).
+  The domain interface and host-device expose a larger amount of parameters
+  which are not exposed through the VMI interface spec.
+  Plugin authors may populate other domain parameters if needed, taking
+  the values as hard-coded or from the VMI object (including annotation).
+
+### Sidecar Artifacts
+
+The expected artifacts include:
+
+- Documentation on how to register it to Kubevirt and what it does.
+- A container image that includes the logic of a network binding plugin sidecar.
+- Optionally, a link to a registry that contains the container image.
```

Comment:
I think these should go together, you provide a container image by persisting it in a registry and providing the  image tag that include the registry.

Reply:
I do not think a link to a registry is mandatory.

It is one thing to provide the sources, another to provide the raw image and yet another to provide it through a registry.

I will add a note about pushing to a registry in order to make it available for consumption in a cluster.